ui/src/app/components/sharded-dataflow-migration-details-form/sharded-dataflow-migration-details-form.component.html

```diff
@@ -83,11 +78,6 @@
                     <input matInput placeholder="Connection profile name" type="text" formControlName="newSourceProfile"
                         required="true" />
                 </mat-form-field>
-                <mat-form-field appearance="outline">
-                    <mat-label>Data Shard Id</mat-label>
-                    <input matInput placeholder="Data ShardId" type="text" formControlName="dataShardId"
-                        required="true" />
-                </mat-form-field>
```

Comment:
Is it possible to generate a `dataShardId` and display it on the UI  in a label instead of removing it? We can also add a toolTip on what this randomly generated string means. 

For the UI, the generated `datashardId` will be passed from the UI, and in the case of CLI, if it is empty, one will be auto-generated using the `generateName` changes you have made above. That sounds reasonable.

This will help the user in understanding the generated resources result later on, since otherwise a bunch of jobs will be randomly listed with no direct way to understanding which job contains which set of logical shards.

Reply:
done